src/Compilers/Core/Portable/Operations/ControlFlowGraphBuilder.cs

```diff
@@ -6565,6 +6565,29 @@ public override IOperation VisitDelegateCreation(IDelegateCreationOperation oper
                 operation.Syntax, operation.Type, operation.ConstantValue, IsImplicit(operation));
         }
 
+        public override IOperation VisitFromEndIndexOperation(IFromEndIndexOperation operation, int? argument)
+        {
+            return new FromEndIndexOperation(operation.IsLifted, operation.IsImplicit, semanticModel: null, operation.Syntax, operation.Type, Visit(operation.Operand), operation.Symbol);
+        }
+
+        public override IOperation VisitRangeOperation(IRangeOperation operation, int? argument)
+        {
+            if (!(operation.LeftOperand is null))
+            {
+                PushOperand(Visit(operation.LeftOperand));
+            }
+
+            IOperation visitedRightOperand = null;
+            if (!(operation.RightOperand is null))
+            {
+                visitedRightOperand = Visit(operation.RightOperand);
+            }
+
+            IOperation visitedLeftOperand = operation.LeftOperand is null ? null : PopOperand();
```

Comment:
>PopOperand [](start = 83, length = 10)

Just curious: what is a scenario where this PushOperand/PopOperand gymnastics would matter (ie. produce a different  output)? I'm assuming it would be a scenario with await.

In the tests so far, that didn't seem observable. Can you add one?

Reply:
This is for when there is some kind of flow control in the first operand, which will result in the operand being replaced with a capture. I'm specifically not adding comprehensive tests here: I'm only adding enough implementation to ensure that the IOperation test hook does not crash.

---
In reply to: [238908725](https://github.com/dotnet/roslyn/pull/31546#discussion_r238908725) [](ancestors = 238908725)